youtube_transcript_api/_cli.py

```diff
@@ -26,19 +25,24 @@ def run(self):
         transcripts = []
         exceptions = []
 
+        Formatter = formats.get_formatter(parsed_args.format)
+
         for video_id in parsed_args.video_ids:
             try:
-                transcripts.append(self._fetch_transcript(parsed_args, proxies, cookies, video_id))
+                transcript = self._fetch_transcript(
+                    parsed_args, proxies, cookies, video_id)
+                transcripts.append(Formatter.format(transcript))
             except Exception as exception:
                 exceptions.append(exception)
 
-        return '\n\n'.join(
+        return ''.join(
```

Comment:
Should be offloaded to the DELIMITER and `.combine` method for the transcripts so they can be customized for a given format. I think I had YAML format on the mind when writing it. To separate multiple "documents" in YAML you would need to separate each document with a `---` instead of newline breaks. Was trying to leave it up to the Formatter to determine desired output basically without hardcoding in newlines which might not be preferred for a specific format.

But I am just realizing exception messages are not accounted for so they will likely not look nice at all if displayed with more than 1 exception or an exception mixed with the output of a transcript. 🤦